sway-core/src/semantic_analysis/ast_node/declaration/impl_trait.rs

```diff
@@ -1033,12 +1054,13 @@ fn check_for_unconstrained_type_parameters(
 
 fn handle_supertraits(
     mut ctx: TypeCheckContext,
+    engines: &Engines,
```

Comment:
This goes for all other instances of it but what's the point of passing an explicit `Engines` argument when it's already part of the `TypeCheckContext`?

Reply:
Well then sorry about the mixed messages, but I think cloning is probably better here. This entire PR is about introducing some acceptable runtime overhead for clarity, and this makes things more confusing rather than less so.